app.rb

```diff
@@ -1,8 +1,70 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+
 class RockPaperScissors < Sinatra::Base
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  enable :sessions
+
   get '/test' do
     'test page'
   end
 
+  get '/' do
+    erb(:index)
+  end
+
+  post '/name' do
+    session[:player1] = params[:player1]
+    redirect('/game')
+  end
+
+  get '/game' do
+    @player1 = session[:player1]
+    erb(:game)
+  end
+
+  post '/weapon' do
+    session[:weapon] = params[:weapon]
+    redirect('/result')
+  end
+
+  def computer_weapon
+    ['Rock', 'Paper', 'Scissors'].sample
```

Comment:
may be worth considering moving game logic to it's own file, to leave your app controller to handle only routing. Maybe class extraction?